charts/docker-hub-rate-limit-exporter-chart/templates/deployment.yaml

```diff
@@ -1,55 +1,59 @@
+{{- if .Values.deployment.enabled }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "docker-hub-rate-limit-exporter.fullname" . }}
   labels:
     {{- include "docker-hub-rate-limit-exporter.labels" . | nindent 4 }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: {{ .Values.deployment.replicaCount }}
   selector:
     matchLabels:
       {{- include "docker-hub-rate-limit-exporter.selectorLabels" . | nindent 6 }}
-      {{- if .Values.podLabels }}
-      {{- toYaml .Values.podLabels | nindent 6 }}
-      {{- end }}
   template:
     metadata:
+      labels:
+        {{- include "docker-hub-rate-limit-exporter.selectorLabels" . | nindent 8 }}
       annotations:
         checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
         checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
-        {{- with .Values.podAnnotations }}
-        {{- toYaml . | nindent 8 }}
+        {{- if .Values.podAnnotations }}
```

Comment:
Is there a reason to change from `with` to ìf` structure for podAnnontations? I have nothing againt it but I would like to understand why you changed it.